# HG changeset patch
# User Rik <rik@octave.org>
# Date 1462396475 25200
#      Wed May 04 14:14:35 2016 -0700
# Node ID 1d66aa16e338905932786e142469968ae933db39
# Parent  5e083d07ba35f423983e191bbe37ebecca4d8043
# Parent  e47eb34f1b05d23014f698c6645f32961475d30d
maint: Periodic merge of stable to default.

diff --git a/scripts/general/sortrows.m b/scripts/general/sortrows.m
--- a/scripts/general/sortrows.m
+++ b/scripts/general/sortrows.m
@@ -18,19 +18,35 @@
 ## <http://www.gnu.org/licenses/>.
 
 ## -*- texinfo -*-
 ## @deftypefn  {} {[@var{s}, @var{i}] =} sortrows (@var{A})
 ## @deftypefnx {} {[@var{s}, @var{i}] =} sortrows (@var{A}, @var{c})
 ## Sort the rows of the matrix @var{A} according to the order of the columns
 ## specified in @var{c}.
 ##
-## If @var{c} is omitted, a lexicographical sort is used.  By default ascending
-## order is used however if elements of @var{c} are negative then the
-## corresponding column is sorted in descending order.
+## By default (@var{c} omitted, or a particular column unspecified in @var{c})
+## an ascending sort order is used.  However, if elements of @var{c} are
+## negative then the corresponding column is sorted in descending order.  If
+## the elements of @var{A} are strings then a lexicographical sort is used.
+##
+## Example: sort by column 2 in descending order, then 3 in ascending order
+##
+## @example
+## @group
+## x = [ 7, 1, 4;
+##       8, 3, 5;
+##       9, 3, 6 ];
+## sortrows (x, [-2, 3])
+##    @result{} 8  3  5
+##       9  3  6
+##       7  1  4
+## @end group
+## @end example
+##
 ## @seealso{sort}
 ## @end deftypefn
 
 ## Author: Daniel Calvelo, Paul Kienzle
 ## Adapted-by: jwe
 
 function [s, i] = sortrows (A, c)
 
